python/mspasspy/db/database.py

```diff
@@ -12,31 +11,55 @@
 
 import bson.errors
 import bson.objectid
+import dask.bag as daskbag
+from pymongo import MongoClient
 import gridfs
 import pymongo
 import numpy as np
 from array import array
 
 from mspasspy.ccore.seismic import (BasicTimeSeries,
-                            TimeSeries,
-                            Seismogram,
-                            CoreSeismogram,
-                            CoreTimeSeries,
-                            TimeReferenceType,
-                            TimeSeries,
-                            DoubleVector)
+                                    TimeSeries,
+                                    Seismogram,
+                                    CoreSeismogram,
+                                    CoreTimeSeries,
+                                    TimeReferenceType,
+                                    TimeSeries,
+                                    DoubleVector,
+                                    TimeSeriesEnsemble,
+                                    SeismogramEnsemble)
 from mspasspy.ccore.utility import (MetadataDefinitions,
                                     Metadata,
                                     dmatrix,
                                     ErrorLogger,
                                     ErrorSeverity, ProcessingHistory)
-from mspasspy.db.schema import MetadataSchema
-
+from mspasspy.db.schema import DatabaseSchema, MetadataSchema
 
 from obspy import Inventory
 from obspy import UTCDateTime
 from obspy import Catalog
 
+
+def _read_distributed_data(client_arg, db_name, id, collection, load_history=True, metadata_schema=None):
+    client = MongoClient(client_arg)
+    db = Database(client, db_name)
+    return db.read_data(id, collection, load_history, metadata_schema)
```

Comment:
why are these not covered (as reported by Codecov)?

Reply:
Codecov is behaving wired (all the time). This is already covered.

Reply:
Just a note here. It seems it is a known issue that Codecov cannot recognize something run by a Spark executor.